src/pyscheduling/JS/JmCmax.py

```diff
@@ -281,11 +281,13 @@ def shifting_bottleneck(instance : JmCmax_Instance):
         Returns:
             RootProblem.SolveResult: SolveResult of the instance by the method
         """
+        startTime = perf_counter()
         solution = JobShop.JobShopSolution(instance)
         graph = JobShop.Graph(instance.P)
         Cmax = graph.critical_path()
         remaining_machines = list(range(instance.m))
         scheduled_machines = []
+        precedence_constraints = []
```

Comment:
Can you please add a comment of how precedence is represented (tuple of two job ids)